QPM/SymLinker/Linker.cs

```diff
@@ -0,0 +1,89 @@
+﻿using SymLinker.LinkCreators;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+
+// Stolen from https://github.com/huffSamuel/SymLinker
+// Cleaned up by Fern, because the old code was horrible
+namespace SymLinker
+{
+    public class Linker
+    {
+        private readonly ISymLinkCreator? _linker;
+
+        /// <summary>
+        /// Creates a new Symbolic Link Creator
+        /// </summary>
+        public Linker()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                _linker = new WindowsSymLinkCreator();
+            }
+            // TODO: Implement
+            // else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            // {
+            //     _linker = new LinuxSymLinkCreator();
+            // }
+            // else
+            // {
+            //     _linker = new OSXSymLinkCreator();
+            // }
+        }
+
+        public bool IsValid() => _linker != null;
```

Comment:
Where's my pattern matching: `_linker is not null` :(

Reply:
Reason for that is because it doesn't call the (potentially) overridden != operator